### PR TITLE
Add http timeout to reporter

### PR DESCRIPTION
**Pull request by bcaglayan**
Add http timeout to reporter

### CODE REVIEW

**Review by aoozdemir — APPROVED** [review]

**Comment by serkan-ozal** [comment]
@batuhancaglayan there are reporter related failing tests

File: src/Reporter.ts [hunk 1]
```diff
@@ -129,6 +134,11 @@ class Reporter {
                     const socket: net.Socket = net.createConnection(options.port as number, options.hostname);
                     socket.setNoDelay(true);
                     socket.setKeepAlive(true);
+
+                    socket.setTimeout(REPORTER_HTTP_TIMEOUT, () => {
+                        ThundraLogger.debug('<Reporter> Reporter socket timeout');
```
Comment (serkan-ozal) [inline src/Reporter.ts:139]: I think, log level should be at error level and log message should mention about possible public internet connectivity issue

Comment (serkan-ozal) [reply src/Reporter.ts:139]:
```js
ThundraLogger.error('<Reporter> Reporter socket timeout. Please be sure that your application has access to public internet.');
```

File: src/Reporter.ts [hunk 2]
```diff
@@ -242,6 +252,11 @@ class Reporter {
                 ? request = https.request(this.requestOptions, responseHandler)
                 : request = http.request(this.requestOptions, responseHandler);
 
+            request.setTimeout(REPORTER_HTTP_TIMEOUT, () => {
+                ThundraLogger.debug('<Reporter> Reporter request timeout');
+                return reject({});
```
Comment (serkan-ozal) [inline src/Reporter.ts:257]:
```js
ThundraLogger.error('<Reporter> Reporter socket timeout. Please be sure that your application has access to public internet.');
```

File: src/Reporter.ts [hunk 3]
```diff
@@ -242,6 +252,11 @@ class Reporter {
                 ? request = https.request(this.requestOptions, responseHandler)
                 : request = http.request(this.requestOptions, responseHandler);
 
+            request.setTimeout(REPORTER_HTTP_TIMEOUT, () => {
+                ThundraLogger.debug('<Reporter> Reporter request timeout');
```
Comment (serkan-ozal) [inline src/Reporter.ts:256]: I think, log level should be at error level and log message should mention about possible public internet connectivity issue

```js
ThundraLogger.error('<Reporter> Reporter socket timeout. Please be sure that your application has access to public internet.');
```

File: src/Reporter.ts (hunk 2 above)
Comment (serkan-ozal) [inline src/Reporter.ts:257]: Should we pass an error to the reject?

**Review by aoozdemir — APPROVED** [review]

**Review by serkan-ozal — APPROVED** [review]

**Pull request merged**